tools/statplot.py

```diff
@@ -1,229 +1,305 @@
+import warnings  # noqa: E402
+
+rc('text', usetex=False)
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description='GUI for Fluidity .stat outputs\n\nWhen the window is '
+    + 'displayed, press the following keys for additional options:\n'
+    + '- x, y -> Change the scale of the x, y axis (automatically chooses '
+    + 'between linear, log and symlog).\n- l -> Change the representation of '
+    + 'the data (solid line or markers).\n- r -> Reload the .stat file. Only '
+    + 'relevant for simulations that are still running.\n- Esc -> Release '
+    + 'focus on text boxes.\n- q -> Exit the GUI.')
+parser.add_argument('statfile', nargs='+', help='path to the .stat file')
+args = parser.parse_args(sys.argv[1:])
+
+
+class StatplotWindow(Gtk.Window):
+    def __init__(self, statfile):
+        Gtk.Window.__init__(self)
+        self.connect('key-press-event', self.KeyPressed)
+        self.set_border_width(8)
+        self.set_default_size(1600, 900)
+        self.set_position(Gtk.WindowPosition.CENTER)
+        self.set_title(statfile[-1])
+        self.statfile = statfile
+        self.vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=8)
+        self.vbox.set_homogeneous(False)
+        self.add(self.vbox)
+        hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=8)
+        hbox.set_homogeneous(True)
+        self.vbox.pack_end(hbox, False, False, 0)
+        self.entries = self.ReadData(statfile)
+        self.xCombo = Gtk.ComboBoxText.new_with_entry()
+        self.xCombo.set_wrap_width(3)
+        self.yCombo = Gtk.ComboBoxText.new_with_entry()
+        self.yCombo.set_wrap_width(3)
+        self.PopulateCombo('load')
+        self.IniCompl(self.xCombo)
+        self.IniCompl(self.yCombo)
+        self.xCombo.connect('changed', self.ComboChangedX)
+        self.xCombo.connect('key-release-event', self.ReleaseFocus)
+        self.yCombo.connect('changed', self.ComboChangedY)
+        self.yCombo.connect('key-release-event', self.ReleaseFocus)
+        hbox.pack_start(self.xCombo, True, True, 0)
+        hbox.pack_start(self.yCombo, True, True, 0)
+        self.fig, self.ax = plt.subplots(nrows=1, ncols=1, num=0,
+                                         figsize=(14, 7))
+        self.canvas = FigureCanvas(self.fig)
+        self.canvas.set_can_focus(True)
+        self.vbox.pack_start(self.canvas, True, True, 0)
+        self.PlotType = 'line'
+        self.PlotData('create', self.PlotType, 'linear', 'linear')
+        self.toolbar = NavigationToolbar(self.canvas, self)
+        self.vbox.pack_start(self.toolbar, False, False, 0)
+
+    def ComboChangedX(self, widget):
+        complOptions = self.PopulateCompletion(self.xCombo.get_active_text())
+        dynModel = Gtk.ListStore.new([str])
+        for option in complOptions:
+            dynModel.append([option])
+        self.xCombo.get_child().get_completion().set_model(dynModel)
```

Comment:
I don't really understand this? You've already set up completion on the ComboBox, so adding a new Model on the Entry every time text is changed seems needless? Remove lines 77-81, and you still have perfectly functioning completion.

Reply:
Entirely right, thank you. Completion is actually doing everything by itself, I guess I wanted to help it, but as you mentioned, needless.